pkg/apis/garden/validation/validation.go

```diff
@@ -1249,11 +1249,20 @@ func validateCloud(cloud garden.Cloud, fldPath *field.Path) field.ErrorList {
 		if len(gcp.Networks.Workers) != zoneCount {
 			allErrs = append(allErrs, field.Invalid(gcpPath.Child("networks", "workers"), gcp.Networks.Workers, "must specify as many workers networks as zones"))
 		}
+
 		workerCIDRs := make([]cidrvalidation.CIDR, 0, len(gcp.Networks.Workers))
 		for i, cidr := range gcp.Networks.Workers {
 			workerCIDRs = append(workerCIDRs, cidrvalidation.NewCIDR(cidr, gcpPath.Child("networks", "workers").Index(i)))
 		}
 
+		if gcp.Networks.Internal != nil {
+			internalCIDR := make([]cidrvalidation.CIDR, 0, 1)
+			internalCIDR = append(internalCIDR, cidrvalidation.NewCIDR(*gcp.Networks.Internal, gcpPath.Child("networks", "internal")))
+			allErrs = append(allErrs, validateCIDRParse(internalCIDR...)...)
+			allErrs = append(allErrs, validateCIDROVerlap([]cidrvalidation.CIDR{pods, services}, internalCIDR, false)...)
+			allErrs = append(allErrs, validateCIDROVerlap([]cidrvalidation.CIDR{nodes}, internalCIDR, false)...)
+		}
```

Comment:
I guess we should also validate that none of the `workers` CIDRs overlap with the `internal` CIDR, right?

Reply:
Can we have a test for this as well please?
Also, please squash your commits after that. Rest looks good I guess